src/data/pnat/basic.lean

```diff
@@ -349,4 +350,122 @@ theorem gcd_mul_lcm (n m : ℕ+) : (gcd n m) * (lcm n m) = n * m :=
 
 def prime (p : ℕ+) : Prop := (p : ℕ).prime
 
+/-- Two pnats are coprime if their gcd is 1. -/
+def coprime (m n : ℕ+) : Prop := m.gcd n = 1
+
+@[simp]
+lemma eq_iff {m n : ℕ+} :
+(m : ℕ) = ↑n ↔ m = n := by { split, apply eq, intro h, rw h }
+
+@[simp]
+lemma coprime_coe {m n : ℕ+} : nat.coprime ↑m ↑n ↔ m.coprime n :=
+by { unfold coprime, unfold nat.coprime, rw ← eq_iff, simp }
+
+lemma coprime.mul {k m n : ℕ+} : m.coprime k → n.coprime k → (m * n).coprime k :=
+by { repeat {rw ← coprime_coe}, rw mul_coe, apply nat.coprime.mul }
+
+lemma gcd_eq_left_iff_dvd {m n : ℕ+} : m ∣ n ↔ m.gcd n = m :=
+by { rw dvd_iff, rw nat.gcd_eq_left_iff_dvd, rw ← eq_iff, simp }
+
+lemma coprime.gcd_mul_right_cancel (m : ℕ+) {n k : ℕ+} :
```

Comment:
There are "permutations" of this one as well. Are they proven for `nat.coprime`? If so, I would like to have them for `pnat` as well.